### PR TITLE
New release

### PR DESCRIPTION
**Pull request by tonyhb**

### CODE REVIEW

File: pages/blog/_posts/ai-orchestration-with-agentkit-step-ai.mdx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    return response.choices;
+  }
+);
+```
+
+## Start building and provide your feedback
+
+As part of our early access release, we’re inviting you and your team to share your thoughts, experiences, and suggestions. Your feedback is invaluable in shaping the future of our APIs to better serve your needs. Sign-up for early access here so we can stay in touch.
+
+We’re excited to see what you build with `step.ai`!
```
Comment (tonyhb) [inline pages/blog/_posts/ai-orchestration-with-agentkit-step-ai.mdx:127]:
```suggestion
We’re excited to see what you build with AgentKit and `step.ai`!
```

File: pages/blog/_posts/ai-orchestration-with-agentkit-step-ai.mdx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    // This calls `generateText` with the given arguments, adding AI observability,
+    // metrics, datasets, and monitoring to your calls.
+    const { text } = await step.ai.wrap("using-vercel-ai", generateText, {
+      model: openai("gpt-4-turbo"),
+      prompt: "What is love?"
+    });
+
+  }
+);
+```
```
Comment (djfarrelly) [inline pages/blog/_posts/ai-orchestration-with-agentkit-step-ai.mdx:92]: A screenshot of what you get for wrapping would be ideal - we're doing a bit of telling, not showing with the "with steps and AI observability" bit

File: pages/docs/agent-kit/ai-agent-network-state-routing.mdx [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+- The agents that the network can access
+- The network’s state, including past messages and a key value store (read more below)
+- The network’s router, which chooses whether to quit or the next agent to run in the loop (read more below)
+
+It also has an optional default model, which will be used when your Agents have no model provided, and a setting to cap the number of model calls via `maxIter`.
+
+Here's a simple example:
+
+```tsx
+// Create a network with two agents.
```
Comment (tonyhb) [inline pages/docs/agent-kit/ai-agent-network-state-routing.mdx:20]:
```suggestion
import { createNetwork, agenticOpenai as openai } from "@inngest/agent-kit";

// Create a network with two agents.
```

File: pages/docs/agent-kit/ai-agent-network-state-routing.mdx [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+const network = createNetwork({
+  agents: [searchAgent, summaryAgent],
+  defaultModel: model,  // Optional: used for routing and agents if they have no model
```
Comment (tonyhb) [inline pages/docs/agent-kit/ai-agent-network-state-routing.mdx:23]:
```suggestion
  defaultModel: openai({ model: "gpt-4o", step }),  // Optional: used for routing and agents if they have no model
```

File: pages/blog/_posts/ai-orchestration-with-agentkit-step-ai.mdx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+    // The response is also strongly typed depending on the model.
+    return response.choices;
+  }
+);
+```
+
+## Start building and provide your feedback
+
+As part of our early access release, we’re inviting you and your team to share your thoughts, experiences, and suggestions. Your feedback is invaluable in shaping the future of our APIs to better serve your needs. Sign-up for early access here so we can stay in touch.
```
Comment (djfarrelly) [inline pages/blog/_posts/ai-orchestration-with-agentkit-step-ai.mdx:125]:
```suggestion
As part of our early access release, we’re inviting you and your team to share your thoughts, experiences, and suggestions. Your feedback is invaluable in shaping the future of our APIs to better serve your needs. [Sign-up for early access here](/ai/early-access?ref=blog-ai-orchestration-with-agentkit-step-ai) so we can stay in touch.
```

File: pages/docs/agent-kit/ai-agent-network-state-routing.mdx [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+const exists = network.state.kv.has("user-name");
+```
+
+Common uses for the KV store include:
+
+- Storing intermediate results that other agents might need within lifecycles
+- Storing user preferences or context
+- Passing data between tools and agents
+
+Here’s a tool which stores files created by a coding agent the network’s key-value state:
```
Comment (djfarrelly) [inline pages/docs/agent-kit/ai-agent-network-state-routing.mdx:124]: Maybe callout that this is using Zod - most people probably know when they see `z`, but if they don't it's confusing

File: pages/docs/agent-kit/ai-agent-network-state-routing.mdx [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+</Callout>
+
+A router is a function that gets called after each agent runs, which decides whether to:
+
+1. Stop the network (by returning `undefined`)
+2. Call another agent (by returning an `Agent`)
+
+The routing function gets access to everything it needs to make this decision:
+
+```jsx
```
Comment (djfarrelly) [inline pages/docs/agent-kit/ai-agent-network-state-routing.mdx:167]:
```suggestion
```ts
```

File: pages/docs/agent-kit/ai-agent-network-state-routing.mdx [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+
+Common uses for the KV store include:
+
+- Storing intermediate results that other agents might need within lifecycles
+- Storing user preferences or context
+- Passing data between tools and agents
+
+Here’s a tool which stores files created by a coding agent the network’s key-value state:
+
+```jsx
```
Comment (djfarrelly) [inline pages/docs/agent-kit/ai-agent-network-state-routing.mdx:126]:
```suggestion
```ts
```

File: pages/docs/agent-kit/ai-agent-network-state-routing.mdx [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+  lastResult?: InferenceResult; // The last agent's response
+}
+```
+
+### Code-based routers (supervised routing)
+
+The simplest way to route is to write code that makes decisions.  Here's an example that routes between a classifier and a writer:
+
+```jsx
+
```
Comment (djfarrelly) [inline pages/docs/agent-kit/ai-agent-network-state-routing.mdx:181]:
```suggestion
```ts
```

File: pages/docs/agent-kit/ai-agents-tools.mdx [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+    // content has already been produced in an agentic workflow.
+    const files: Record<string, string> | undefined = network?.state.kv.get("files")
+    if (files === undefined) {
+      return systemPrompt;
+    }
+
+    // There are files present in the network's state, so add them to the promp to help
+    // provide previous context automatically.
+		let prompt = systemPrompt + "The following code already exists:"
+		for (const [name, contents] of Object.entries(record)) {
```
Comment (djfarrelly) [inline pages/docs/agent-kit/ai-agents-tools.mdx:107]:
```suggestion
    let prompt = systemPrompt + "The following code already exists:"
    for (const [name, contents] of Object.entries(record)) {
```

File: pages/docs/agent-kit/ai-agents-tools.mdx [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+When you create an agent you can specify any number of tools that the agent can use. Tools follow the standard formats that OpenAI and Anthropic provide: a name and a description, plus typed parameters.
+
+In AgentKit, you also define a `handler` function which is called when the tool is invoked.  Because AgentKit runs in the backend (on your own infrastructure) these handlers can run almost any code you define.
+
+
+### Complex agents with tools
+
+A more complex agent used in a network defines a description, lifecycle hooks, tools, and a dynamic set of instructions based off of network state:
+
+```jsx
```
Comment (djfarrelly) [inline pages/docs/agent-kit/ai-agents-tools.mdx:80]:
```suggestion
```ts
```

**Review by djfarrelly — APPROVED** [review]

File: pages/blog/_posts/ai-orchestration-with-agentkit-step-ai.mdx [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+    // Create a network of agents with separate tasks and instructions to solve
+    // a specific task, using Inngest steps for orchestration 
+    const network = new Network({
+      agents: [navigator, classifier, summarizer],
+      defaultModel: agenticOpenai({ model: "gpt-4o", step }),
+    })
+    
+    const input = "Classify then summarize the latest 10 blog posts on https://www.deeplearning.ai/blog/"
+    
+    const result = await network.run(, ({ network }) => {
```
Comment (tonyhb) [inline pages/blog/_posts/ai-orchestration-with-agentkit-step-ai.mdx:42]:
```suggestion
    const result = await network.run(input, ({ network }) => {
```

File: pages/blog/_posts/ai-orchestration-with-agentkit-step-ai.mdx [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+date: 2024-11-22
+author: Tony Holdstock-Brown
+category: product-updates
+---
+
+Building AI applications that actually work in production is surprisingly complex.  Your model calls need to be reliable, you need to store history across timeouts, retries, and multiple model calls,  and you need to handle flow control successfully, all while iterating on things like your actual prompts and RAG approach.  To be successful, this requires careful orchestration.  This is why we built Inngest — the developer-first platform that lets handles orchestration at scale, so you can focus on your product without worrying about technical details.
+
+Today, we're excited to announce two new AI-native features:
+
+- [AgentKit](https://github.com/inngest/agent-kit), the fastest way to develop, test, and ship complex AI workflows to production, in early access
```
Comment (tonyhb) [inline pages/blog/_posts/ai-orchestration-with-agentkit-step-ai.mdx:15]:
```suggestion
- [AgentKit](/docs/agent-kit/overview), the fastest way to develop, test, and ship complex AI workflows to production, in early access
```

File: pages/docs/agent-kit/ai-agent-network-state-routing.mdx (hunk 6 above)
Comment (tonyhb) [inline pages/docs/agent-kit/ai-agent-network-state-routing.mdx:124]:
```suggestion
Here’s a tool described using a Zod schema which stores files created by a coding agent the network’s key-value state:
```

**Pull request merged**